contrib/camel/pkg/apis/sources/v1alpha1/camelsource_types.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis/duck"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CamelSource is the Schema for the camelsources API
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:categories=all,knative,eventing,sources
+type CamelSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CamelSourceSpec   `json:"spec,omitempty"`
+	Status CamelSourceStatus `json:"status,omitempty"`
+}
+
+// Check that CamelSource can be validated and can be defaulted.
+var _ runtime.Object = (*CamelSource)(nil)
+
+// Check that CamelSource implements the Conditions duck type.
+var _ = duck.VerifyType(&CamelSource{}, &duckv1alpha1.Conditions{})
+
+const (
+	// CamelSourceConditionReady has status True when the CamelSource is ready to send events.
+	CamelConditionReady = duckv1alpha1.ConditionReady
+
+	// CamelConditionSinkProvided has status True when the CamelSource has been configured with a sink target.
+	CamelConditionSinkProvided duckv1alpha1.ConditionType = "SinkProvided"
+
+	// CamelConditionDeployed has status True when the CamelSource has had it's deployment created.
+	CamelConditionDeployed duckv1alpha1.ConditionType = "Deployed"
+)
+
+var camelCondSet = duckv1alpha1.NewLivingConditionSet(
+	CamelConditionSinkProvided,
+	CamelConditionDeployed,
+)
+
+// CamelSourceSpec defines the desired state of CamelSource
+type CamelSourceSpec struct {
+	// Source is the reference to the integration flow to run.
+	Source CamelSourceOriginSpec `json:"source"`
+
+	// ServiceAccountName is the name of the ServiceAccount to use to run this
+	// source.
```

Comment:
What is the default? I assume default :D

Reply:
Yes ;)